src/librustc_typeck/check/method/suggest.rs

```diff
@@ -70,7 +70,8 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                                rcvr_ty: Ty<'tcx>,
                                item_name: ast::Name,
                                rcvr_expr: Option<&hir::Expr>,
-                               error: MethodError<'tcx>) {
+                               error: MethodError<'tcx>,
+                               args: Option<&'gcx [hir::Expr]>) {
```

Comment:
In the `None` case here, could you pass an empty slice, and then not require `Option<...>`? If not, you should add a comment, explaining when and why to use `None`.